### PR TITLE
V2X: Update obu interface

### PR DESCRIPTION
**Pull request by QIU1995NONAME**

### CODE REVIEW

File: modules/v2x/v2x_proxy/obu_interface/grpc_interface/grpc_server.cc [hunk 1]
```diff
@@ -136,6 +150,21 @@ void GrpcServerImpl::GetMsgFromGrpc(std::shared_ptr<ObuRsi> *ptr) {
   (*ptr)->CopyFrom(latest_rsi_);
 }
 
+void GrpcServerImpl::GetMsgFromGrpc(
+    std::shared_ptr<::apollo::v2x::V2XObstacles> *ptr) {
```
Comment (lx18233184051) [inline modules/v2x/v2x_proxy/obu_interface/grpc_interface/grpc_server.cc:154]: since have used std::shared_ptr, why still use * ?

Comment (QIU1995NONAME) [reply modules/v2x/v2x_proxy/obu_interface/grpc_interface/grpc_server.cc:154]: ```apollo.sh lint``` don't allow to use non-conf refs, also in order to be consistent with the existing code style.

File: modules/v2x/v2x_proxy/obu_interface/grpc_interface/grpc_server.cc [hunk 2]
```diff
@@ -72,6 +72,20 @@ grpc::Status GrpcServerImpl::SendV2xRSI(grpc::ServerContext * /* context */,
   AINFO << "SendV2xRSI response success.";
   return Status::OK;
 }
+grpc::Status GrpcServerImpl::SendPerceptionObstacles(
+    grpc::ServerContext *, const apollo::v2x::V2XObstacles *request,
+    StatusResponse *response) {
+  ADEBUG << "Received SendPerceptionObstacles request from client! \n";
```
Comment (lfcarol) [inline modules/v2x/v2x_proxy/obu_interface/grpc_interface/grpc_server.cc:78]: Remove '\n'

File: modules/v2x/v2x_proxy/obu_interface/grpc_interface/grpc_server.h [hunk 3]
```diff
@@ -58,15 +58,23 @@ class GrpcServerImpl final : public ::apollo::v2x::ObuToCar::Service {
...
   grpc::Status SendObuAlarm(grpc::ServerContext *context,
                             const ::apollo::v2x::ObuAlarm *request,
-                            ::apollo::v2x::StatusResponse *response);
+                            ::apollo::v2x::StatusResponse *response) override;
+
+  /* function that send perception obstacles through grpc
+     @param input perception obstacles grpc request
+     @param output grpc response
+    */
+  grpc::Status SendPerceptionObstacles(  //
```
Comment (lfcarol) [inline modules/v2x/v2x_proxy/obu_interface/grpc_interface/grpc_server.h:75]: Remove '//'

**Review by lx18233184051 — APPROVED** [review]

**Pull request merged**